omnigibson/object_states/touching.py

```diff
@@ -21,4 +23,8 @@ def _get_value(self, other):
         elif other.prim_type == PrimType.CLOTH:
             return self._check_contact(self.obj, other)
         else:
-            return self._check_contact(other, self.obj) and self._check_contact(self.obj, other)
+            # Use optimized check for rigid bodies
+            return RigidContactAPI.in_contact(
+                prim_paths_a=[link.prim_path for link in self.obj.links.values()],
+                prim_paths_b=[link.prim_path for link in other.links.values()],
+            )
```

Comment:
Can we discuss this in person? I guess I am not sure when we should use ContactBodies vs RigidContactAPI.